tendrl/commons/jobs/__init__.py

```diff
@@ -96,6 +111,17 @@ def run(self):
     def stop(self):
         pass
 
+    def read_job(self, key):
+        result = {}
+        job = tendrl_ns.etcd_orm.client.read(key)
+        if hasattr(job, '_children'):
+            for item in job._children:
```

Comment:
Accessing private variable "_children" is not at all the ideal way to deal with this, you could have looked at the python etcd client API (https://github.com/jplana/python-etcd/blob/0.4.4/src/etcd/__init__.py#L90)

Please use the "leaves" property to iterate through to the job attributes and use the python-etcd client provided help attributes for EtcdResult